codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithydafny/DafnyApiCodegen.java

```diff
@@ -1312,6 +1325,104 @@ private TokenTree OperationMemberValidState(
         )
         .dropEmpty()
         .lineSeparated();
+    } else if (isUnion && member.isRequired()) {
+      // Required Union
+
+      final UnionShape union = memberShape
+        .asUnionShape()
+        .get();
+
+      return TokenTree
+        .of(
+          TokenTree.of("\n && (match %s ".formatted(varName)),
+          TokenTree.of(union
+            .members()
+            .stream()
+            .filter(this::OnlyReferenceStructures)
+            .map(s -> TokenTree.of(
```

Comment:
Suggestion: I am not sure what the line comment should be, 
but there should be a line comment... 
Because Smithy-Dafny is complicated.
```suggestion
            .stream()
            // Reference Structures in a Union need Modifies
            .filter(this::OnlyReferenceStructures)
            .map(s -> TokenTree.of(
```

Reply:
added, but the code moved. Added a slightly different comment, but still something